src/Lykke.Job.CandlesHistoryWriter.Services/HistoryMigration/TradesMigrationManager.cs

```diff
@@ -4,182 +4,92 @@
-                    }
-                    catch (Exception ex)
-                    {
-                        Health.State = TradesMigrationState.Error;
-                        await _log.WriteErrorAsync(nameof(TradesMigrationManager), nameof(DoMigrateAsync), ex);
-                        await _log.WriteInfoAsync(nameof(TradesMigrationManager), nameof(DoMigrateAsync),
-                            $"Migration for {migrationItem.AssetPairId} interrupted due to error. Please, check error log for details.");
-                        return;
-                    }
-                }
+                assetSearchTokens.Add((
+                    AssetPairId: assetPairId,
+                    SearchToken: storedAssetPair.BaseAssetId + storedAssetPair.QuotingAssetId,
+                    ReverseSearchToken: storedAssetPair.QuotingAssetId + storedAssetPair.BaseAssetId));
             }
 
-            Health.State = TradesMigrationState.Finished;
-        }
-
-        private void ExtendStoredCandles(ref TradesCandleBatch current)
-        {
-            var storedCandles =
-                _candlesHistoryRepository.GetCandlesAsync(current.AssetId, current.TimeInterval, TradesCandleBatch.PriceType, current.MinTimeStamp,
-                    current.MaxTimeStamp.AddSeconds((int)current.TimeInterval)).GetAwaiter().GetResult().ToList();
-            if (!storedCandles.Any())
-                return;
+            // Creating a blank health report
+            _tradesMigrationHealthService.Prepare(_sqlQueryBatchSize, preliminaryRemoval, removeByDate);
 
-            for (int i = 0; i < current.CandlesCount; i++)
+            // If there is nothing to migrate, just return "success".
+            if (!assetSearchTokens.Any())
             {
-                var timestamp = current.Candles.Values.ElementAt(i).Timestamp;
-                var stored = storedCandles.FirstOrDefault(s => s.Timestamp == timestamp);
-                if (stored != null)
-                {
-                    current.Candles[timestamp.ToFileTimeUtc()] =
-                        stored.ExtendBy(current.Candles[timestamp.ToFileTimeUtc()]);
-                    storedCandles.Remove(stored);
-                }
+                _tradesMigrationHealthService.State = TradesMigrationState.Finished;
+                _log.WriteInfoAsync(nameof(TradesMigrationManager), nameof(Migrate),
+                        $"Trades migration: None of the requested asset pairs can be stored. Migration canceled.").GetAwaiter().GetResult();
+                return true;
             }
+
+            // We do not parallel the migration of different asset pairs consciously.
+            // If we have no upper date-time limit for migration, we migrate everything.
+            Task.Run(() => 
+                _tradesMigrationService.MigrateTradesCandlesAsync(preliminaryRemoval, removeByDate, assetSearchTokens)
+                    .GetAwaiter()
```

Comment:
You don't need here .GetAwaiter() as you don't wait for a result in the outer code